### PR TITLE
Gh674 [ltcmdhooks] Make patching commands a global operation

### LINKED ISSUE #674: Generic command hooks should be patched in globally
## Brief outline of the bug

Hook data structures are globally manged so adding a generic hook also need to happen globably. 

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{article}
\begingroup
\AddToHook{cmd/textbf/before}{\XXX}  % patching happens later so no longer inside a group
\endgroup

\begin{document}

\begingroup
\AddToHook{cmd/textbf/after}{\YYY}   % patch is undone at group end
\endgroup

\ShowCommand\textbf              % after hook not patched in but "before" hook is
\ShowHook{cmd/textbf/before}
\ShowHook{cmd/textbf/after}

\end{document}
```

## Log file (required) and possibly PDF file  

```
> \textbf=robust macro:
->\protect \textbf  .

> \textbf =\long macro:
#1->\UseHook {cmd/textbf/before}\ifmmode \nfss@text {\bfseries #1}\else \hmode@
bgroup \text@command {#1}\bfseries \check@icl #1\check@icr \expandafter \egroup
 \fi .
<argument> \textbf  
                    
l.13 \ShowCommand\textbf
                                      % after hook not patched in but "befor...

? 

-> The hook 'cmd/textbf/before':
> Code chunks:
>     ---
> Document-level (top-level) code (executed last):
>     -> \XXX 
> Extra code for next invocation:
>     ---
> Rules:
>     ---
> Execution order:
>     ---.
<recently read> }
                 
l.14 \ShowHook{cmd/textbf/before}
                                 
? 

-> The hook 'cmd/textbf/after':
> Code chunks:
>     ---
> Document-level (top-level) code (executed first):
>     -> \YYY 
> Extra code for next invocation:
>     ---
> Rules:
>     ---
> Execution order (after reversal):
>     ---.
<recently read> }
                 
l.15 \ShowHook{cmd/textbf/after}
```

### PR DESCRIPTION
**Pull request by PhelypeOleinik**
Fixes #674

# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

**Comment by PhelypeOleinik** [comment]
Not sure if an ltnews entry is needed here...  One hand is likely enough to count how many people noticed this quirk before.

**Comment by car222222** [comment]
I would say not needed.

File: base/changes.txt [hunk 1]
```diff
@@ -6,6 +6,11 @@ completeness or accuracy and it contains some references to files that
 are not part of the distribution.
 ================================================================================
 
+2021-09-28 Phelype Oleinik  <phelype.oleinik@latex-project.org>
+
+	* ltcmdhooks.dtx (subsection{Patching commands}):
+	Make patching commands a global operation (gh/674).
+
```
Comment (FrankMittelbach) [inline base/changes.txt:13]: would mention what is patched:

Make patching commands with generic cmd hooks a global operation (gh/674)

it is not that each kind of patching is suddenly global

**Review by FrankMittelbach — APPROVED** [review]

File: base/changes.txt (hunk 1 above)
Comment (car222222) [reply base/changes.txt:13]: Make patching of commands . . . 

(more accurate!)

Comment (car222222) [reply base/changes.txt:13]: Also agreeing with Frank's change.

**Pull request merged**